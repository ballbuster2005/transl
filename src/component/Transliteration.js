import React from 'react'
import TextBox from './TextBox/TextBox'
import ActionButton from './ActionButton/ActionButton'
import TranslitModel from './../model/TranslitModel'

class Transliteration extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            Value:     "",
            result:    "",
        }
        this.onPress = this.onPress.bind(this)

    }

    onPress = (label) => {
        const text = String(document.getElementById("text").value)
        this.setState({
            result:     TranslitModel(text)
        })
    }

    render() {
        return (
            <div className="Transliteration">
                <TextBox label="Text: " id="text" value={this.state.Value}></TextBox>
                <ActionButton label="Transliterate" action={this.onPress}></ActionButton>
                <h1>Transliteration: {this.state.result}</h1>
            </div>
        )
    }
}

export default Transliteration;
