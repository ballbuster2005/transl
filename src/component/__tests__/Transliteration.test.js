import React, { createElement } from 'react'
import { create } from 'react-test-renderer'

import Transliteration from './../Transliteration'

// Still makes sense to see if this Component actually renderable
describe("[RENDER] Transliteration â€“ Snapshot test", ()=> {
    test("Matches the snapshot", ()=>{
        const translit = create(<Transliteration/>)
        expect(translit.toJSON()).toMatchSnapshot()
    })
})
