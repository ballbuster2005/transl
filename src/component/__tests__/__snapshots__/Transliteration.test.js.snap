// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Transliteration â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Transliteration"
>
  <div
    class="TextBox"
    className="text"
  >
    <label
      className="inputLabel"
    >
      Text: 
    </label>
    <input
      className="inputValue"
      id="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    class="ActionButton"
    className="ActionButton"
  >
    <button
      class="button"
      onClick={[Function]}
    >
      Transliterate
    </button>
  </div>
  <h1>
    Transliteration: 
    
  </h1>
</div>
`;
